Lib/contextlib.py

```diff
@@ -441,7 +441,16 @@ def __exit__(self, exctype, excinst, exctb):
         # exactly reproduce the limitations of the CPython interpreter.
         #
         # See http://bugs.python.org/issue12029 for more details
-        return exctype is not None and issubclass(exctype, self._exceptions)
+        if exctype is None:
+            return
+        if issubclass(exctype, self._exceptions):
+            return True
+        if issubclass(exctype, ExceptionGroup):
```

Comment:
Was there a specific reason to handle `ExceptionGroup` but not `BaseExceptionGroup`?

Reply:
I doubt it. Want to make a PR?